src/components/views/globals/UpdateCheckBar.js

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2017 Michael Telatynski <7t3chguy@gmail.com>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+'use strict';
+
+import React from 'react';
+import dis from 'matrix-react-sdk/lib/dispatcher';
+import { _t } from 'matrix-react-sdk/lib/languageHandler';
+import PlatformPeg from 'matrix-react-sdk/lib/PlatformPeg';
+import {updateStateEnum} from '../../../vector/platform/VectorBasePlatform';
+import AccessibleButton from 'matrix-react-sdk/lib/components/views/elements/AccessibleButton';
+
+export default React.createClass({
+
+    getInitialState: function() {
+        return {
+            message: _t('Checking for an update...'),
+            done: false,
+        };
+    },
+
+    componentWillMount: function() {
+        PlatformPeg.get().checkForUpdate().done((state) => {
```

Comment:
I'm not hugely keen on componentWillMount having side effects - I'd prefer if the cause & effect was the other way around, ie. firing off an update check causes the bar to appear (if it was a user-triggered one)